[Goh Yijie Jonathan] ip

jgyj123 opened the pull request:
# **Project Duke**
- Swift command line interface
- good user experience
- best robot out there :+1:

## Some of the core features include:
1. Keeping track of tasks using ```list```
2. Marking them as complete
3. Storing of tasks

- [x]  ~~Completed task~~
- [x] Do homework
- [x] Submit the assignment (by 20 Aug 2022)

>This is the best robot I have ever met
This robot was built by [Jonathan](https://github.com/jgyj123)

Here is some random code:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

jgyj123 committed:
Added command echoing (level -1)

jgyj123 committed:
Level 2. Add, List

jgyj123 committed:
Add the ability to mark tasks as done.

jgyj123 committed:
Add support for tracking three types of tasks:
1) Todos
2) Deadlines
3) Events

jgyj123 committed:
Add Automated Text UI Testing

jgyj123 committed:
Add Exception Handling

jgyj123 committed:
Add support for deleting tasks from the list.

jgyj123 committed:
Add support for saving task in hard disk

jgyj123 committed:
Add support for missing data folder

jgyj123 committed:
Add date parsing functionality

jgyj123 committed:
Add parsing for dates

jgyj123 committed:
Revert "Add parsing for dates"

This reverts commit 92d373274bcfaa1b409618a54332fd3c3af73933.

jgyj123 committed:
Revert "Revert "Add parsing for dates""

This reverts commit 1e2205a783e47eccb56151acce5a1d06655b6e5f.

jgyj123 committed:
Revert "Revert "Revert "Add parsing for dates"""

This reverts commit 41fea6ae56f34eb125f22638c2601d7f443b6579.

jgyj123 committed:
Add date parsing

jgyj123 committed:
Resolve bugs after merging branches 7 and 8

jgyj123 committed:
Use more OOP

jgyj123 committed:
Add gradle support

jgyj123 committed:
Add test file skeleton

jgyj123 committed:
Add JUnit tests

jgyj123 committed:
Update marking of finished tasks

jgyj123 committed:
Add JavaDoc for classes and methods

jgyj123 committed:
Follow coding standard

jgyj123 committed:
Enable users to find task by searching for keyword

jgyj123 committed:
Merge branch 'A-JavaDoc'

jgyj123 committed:
Merge branch 'A-CodingStandard'

jgyj123 committed:
Merge branch 'Level-9'

malwaregarry commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+     * @param s Command containing task message and time
+     * @param isCompleted whether the task is completed
+     */
+    public Deadline (String s , boolean isCompleted) {
```
Should this whitespace be removed?

malwaregarry commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,51 @@
+/**
+ * Main java class to create a new Duke object
```
according to the module's coding standard, javadoc comments should end with a fullstop. This is different from git subject headers which states not to end with a fullstop.

malwaregarry commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,99 @@
+        }
+        if (taskAdded) {
+            Task addedTask = tasks.get(tasks.getSize() - 1);
+            System.out.println("Got it. I've added this task: \n" + addedTask.toString() +
```
perhaps the `+` would be nicer at the start of the next line:
```
System.out.println("Got it. I've added this task: \n" + addedTask.toString()
        + "\nNow you have " +  tasks.getSize() +  " tasks in the list.");
```

malwaregarry commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,99 @@
+public class Parser {
+    private Storage s;
+    private Ui ui;
+    private TaskList tasks;
```
perhaps a better variable would be `taskList` as `tasks` might imply a collection of task

malwaregarry commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,34 @@
+/**
+ * Represents an Event that can be created by the user
+ */
```
perhaps it would be more readable just before the class

malwaregarry commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,74 @@
+                if (taskType.equals("T")) {
+                    tasks.add(new Todo("todo " + taskMessage, isTaskCompleted));
+                }
+                else if (taskType.equals("E")) {
```
perhaps to you could start the else clause to the curly bracket to meet the coding standard?
```
if (condition) {
    statements;
} else if (condition) {
    statements;
} else {
    statements;
}
```

malwaregarry commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    /**
+     * functional method to delete a task
```
I believe the first letter in each javadoc comment should be capitalised?

malwaregarry reviewed (commented):
LGTM not much issues. Your code is easy to read and I could only spot some minor coding standard violations.

jgyj123 committed:
Divide classes into packages

jgyj123 committed:
Add GUI to Duke using JavaFX

jgyj123 committed:
Merge branch 'branch-Level-10'

jgyj123 committed:
update entry point of application

jgyj123 committed:
Use Assert feature

There is a need to check that methods in TaskList and Storage class are being passed valid arguments

Let's use assertions to implement these safety checks

Using assertions will help to reduce bugs in the code and increase confidence in the correctness of the code

jgyj123 committed:
Improve code quality of code in parser class

Parse method is too long and hard to read

Let's refactor the parse method

This will help to make the code more readable and scalable

jgyj123 committed:
Merge pull request #1 from jgyj123/A-Assertions

Use Assert feature

There is a need to check that methods in TaskList and Storage class are being passed valid arguments

Let's use assertions to implement these safety checks

Using assertions will help to reduce bugs in the code and increase confidence in the correctness of the code

jgyj123 committed:
Merge branch 'master' into A-CodeQuality

* master:
  Use Assert feature

jgyj123 committed:
Merge pull request #2 from jgyj123/A-CodeQuality

Improve code quality of code in parser class

Parse method is too long and hard to read

Let's refactor the parse method

This will help to make the code more readable and scalable

jgyj123 committed:
Allow users to view deadlines and events on a specified date

jgyj123 committed:
Allow users to indicate time of event and deadline

jgyj123 committed:
Add user guide

jgyj123 committed:
Add line spacing to readme

jgyj123 committed:
Add more line spacing to readme
